### PR TITLE
Add support for meta fields

### LINKED ISSUE #56: Allow to set meta
Would be nice to allow to set meta for each resources. Something like this I imagine:

```
# message_view.ex
defmodule Foo.MessageView do
  def meta
    %{ count: Message.count, unread_count: Message.unread_count_for(params[:id]) }
  end
end
```

Not sure how we can pass the "params" in there though. Any thoughts?

### PR DESCRIPTION
**Pull request by doomspork**
@jeregrine / @sebbean / @gottfrois — feedback welcomed and encouraged!

This closes #56.

### CODE REVIEW

File: lib/jsonapi/serializer.ex [hunk 1]
```diff
@@ -38,6 +38,7 @@ defmodule JSONAPI.Serializer do
       id: view.id(data),
       type: view.type(),
       attributes: view.attributes(data, conn),
+      meta: view.meta(data, conn),
```
Comment (jeregrine) [inline lib/jsonapi/serializer.ex]: I think we should only add meta if view.meta() is not nil? The spec doesn't state if we should send down empty maps or not but if someone isn't setting meta perhaps they'd prefer it not be sent down?

Subsequently we should return nil here. https://github.com/jeregrine/jsonapi/pull/62/files#diff-c44c9da7e2c91d6f0643f4ef73ab636cR88

**Comment by jeregrine** [comment]
Thank you so much for this though! Really appreciate the assistance.

File: lib/jsonapi/serializer.ex (hunk 1 above)
Comment (doomspork) [inline lib/jsonapi/serializer.ex]: @jeregrine I coded this based on your comments in the original issue:

> have a default view.ex meta function that returns an empty %{} and is overrideable.

You want to deviate from that now?

**Comment by doomspork** [comment]
@jeregrine I've gone ahead and changed this to return `nil` and only include `meta` in the event that the value is _not_ `nil`.

**Comment by jeregrine** [comment]
Thank you!!

**Comment by doomspork** [comment]
You're welcome @jeregrine 👍

**Pull request merged**